pkg/controller/certificates/sync.go

```diff
@@ -218,6 +224,16 @@ func (c *Controller) updateSecret(name, namespace string, cert, key []byte) (*ap
 	}
 	secret.Data[api.TLSCertKey] = cert
 	secret.Data[api.TLSPrivateKeyKey] = key
+
+	if secret.Annotations == nil {
+		secret.Annotations = make(map[string]string)
+	}
+
+	secret.Annotations[altNamesAnnotation] = strings.Join(spec.DNSNames, ",")
+	secret.Annotations[commonNameAnnotation] = spec.CommonName
+	secret.Annotations[issuerNameAnnotation] = spec.IssuerRef.Name
+	secret.Annotations[issuerKindAnnotation] = spec.IssuerRef.Kind
```

Comment:
Again, cannot read Kind directly from the Certificate spec as it is an optional field and we don't have defaulting yet due to CRDs (and we haven't created a MutatingWebhook in cert-manager yet)